src/sentry/integrations/slack/actions/notification.py

```diff
@@ -117,12 +127,81 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
                     "attachments": json.dumps(attachments),
                 }
 
+            rule = rules[0] if rules else None
```

Comment:
I wish i could write a test for this, but this is a function in a function

Reply:
is it possible to break out the different parts? maybe make a ticket?

Reply:
Yeah I was thinking the same, will try to capture it in my next PR